ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js

```diff
@@ -25,22 +28,25 @@ function getDescriptionNode(permission, index) {
   );
 }
 
-export default function PermissionsConnectPermissionList({
-  permissions,
-  targetSubjectMetadata,
-}) {
+export default function PermissionsConnectPermissionList({ permissions }) {
   const t = useI18nContext();
+  const snapsMetadata = useSelector(getSnapsMetadata);
+
+  const getSnapName = (id) => {
+    return snapsMetadata[id]?.name ?? stripSnapPrefix(id);
+  };
```

Comment:
Can we extract this and add it to a utils file ? This function is redefined in multiple places

Reply:
Hmm yeah, I'll turn it into a closure